packages/lds-jws2020/src/JsonWebSignature2020.js

```diff
@@ -0,0 +1,165 @@
+   * @param [LDKeyClass] {LDKeyClass} provided by subclass or subclass
+   *   overrides `getVerificationMethod`.
+   *
+   *
+   * This parameter is required for signing:
+   *
+   * @param [signer] {function} an optional signer.
+   *
+   * @param [proofSignatureKey] {string} the property in the proof that will contain the signature.
+   * @param [date] {string|Date} signing date to use if not passed.
+   * @param [key] {LDKeyPair} an optional crypto-ld KeyPair.
+   * @param [useNativeCanonize] {boolean} true to use a native canonize
+   *   algorithm.
+   */
+  constructor({
+    linkedDataSigantureType,
+    linkedDataSignatureVerificationKeyType,
+    alg,
+    LDKeyClass,
+    signer,
+    key,
+    proofSignatureKey,
+    date,
+    useNativeCanonize
+  } = {}) {
+    super({
+      type: linkedDataSigantureType,
+      LDKeyClass,
+      alg,
+      date,
+      useNativeCanonize
+    });
+    this.alg = alg;
+    this.LDKeyClass = LDKeyClass;
+    this.signer = signer;
+    this.requiredKeyType = linkedDataSignatureVerificationKeyType;
+    this.proofSignatureKey = proofSignatureKey || "jws";
+
+    if (key) {
+      const publicKey = key.publicNode();
+      this.verificationMethod = publicKey.id;
+      this.key = key;
+      if (typeof key.signer === "function") {
+        this.signer = key.signer();
+      }
+      if (typeof key.verifier === "function") {
+        this.verifier = key.verifier(key, this.alg, this.type);
+      }
+    }
+
+    if (this.alg === undefined) {
+      this.alg = getRecomendedAlg(key.publicKeyJwk);
+    }
+  }
+
+  /**
+   * @param verifyData {Uint8Array}.
+   * @param proof {object}
+   *
+   * @returns {Promise<{object}>} the proof containing the signature value.
+   */
+  async sign({ verifyData, proof }) {
+    if (!(this.signer && typeof this.signer.sign === "function")) {
+      throw new Error("A signer API has not been specified.");
+    }
+
+    proof[this.proofSignatureKey] = await this.signer.sign({
+      data: verifyData
+    });
+    return proof;
+  }
+
+  /**
+   * @param verifyData {Uint8Array}.
+   * @param verificationMethod {object}.
+   * @param document {object} the document the proof applies to.
+   * @param proof {object} the proof to be verified.
+   * @param purpose {ProofPurpose}
+   * @param documentLoader {function}
+   * @param expansionMap {function}
+   *
+   * @returns {Promise<{boolean}>} Resolves with the verification result.
+   */
+  async verifySignature({ verifyData, verificationMethod, proof }) {
+    let { verifier } = this;
+
+    if (!verifier) {
+      const key = await this.LDKeyClass.from(verificationMethod);
+      verifier = key.verifier(key, this.alg, this.type);
+    }
+    return await verifier.verify({
+      data: Buffer.from(verifyData),
+      signature: proof[this.proofSignatureKey]
+    });
+  }
+
+  async assertVerificationMethod({ verificationMethod }) {
+    if (!jsonld.hasValue(verificationMethod, "type", this.requiredKeyType)) {
```

Comment:
This was in the original version too, but jsonld isn't `required` in this file, should I add that in?

Reply:
yes please, we appear to be missing a unit test for that.